Bazel BUILD fails during analysis phase when python2.7 binary is installed, but python2.7-dev is not.

I ran into a situation where I have `python3` and `python3-dev` installed, but I also happen to have legacy `python2.7` installed. Since python2.7 is present by default on many systems, this is a quite common situation.
 
When I try to run grpc bazel build (even when trying to build just grpc C++), it fails very early with the message below.
I think the issue is that the configurator sees that python2.7 is available, so it tries to configure it, but fails since pyhon2's python.h isn't present (which is fair since I never wanted to use python2 for building anything).

Can we get rid of python2 support in our bazel build entirely? AFAIK our bazel BUILD should only be using python3 at this point, since we deprecated python2.

```
INFO: Repository local_config_python instantiated at:
  /var/local/git/grpc/WORKSPACE:5:10: in <toplevel>
  /var/local/git/grpc/bazel/grpc_deps.bzl:448:21: in grpc_deps
  /var/local/git/grpc/bazel/grpc_python_deps.bzl:61:21: in grpc_python_deps
Repository rule python_configure defined at:
  /var/local/git/grpc/third_party/py/python_configure.bzl:365:35: in <toplevel>
ERROR: An error occurred during the fetch of repository 'local_config_python':
   Traceback (most recent call last):
	File "/var/local/git/grpc/third_party/py/python_configure.bzl", line 347, column 35, in _python_autoconf_impl
		_create_single_version_package(
	File "/var/local/git/grpc/third_party/py/python_configure.bzl", line 304, column 45, in _create_single_version_package
		python_include = _get_python_include(repository_ctx, python_bin)
	File "/var/local/git/grpc/third_party/py/python_configure.bzl", line 253, column 13, in _get_python_include
		_execute(
	File "/var/local/git/grpc/third_party/py/python_configure.bzl", line 62, column 14, in _execute
		_fail("\n".join([
	File "/var/local/git/grpc/third_party/py/python_configure.bzl", line 35, column 9, in _fail
		fail("%sPython Configuration Error:%s %s\n" % (red, no_color, msg))
Error in fail: Python Configuration Error: Unable to find Python headers for /usr/bin/python2
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AssertionError: /usr/include/python2.7/Python.h does not exist.
Are Python headers installed? Try installing python-dev or python3-dev on Debian-based systems. Try python-devel or python3-devel on Redhat-based systems.
ERROR: Error fetching repository: Traceback (most recent call last):
	File "/var/local/git/grpc/third_party/py/python_configure.bzl", line 347, column 35, in _python_autoconf_impl
		_create_single_version_package(
	File "/var/local/git/grpc/third_party/py/python_configure.bzl", line 304, column 45, in _create_single_version_package
		python_include = _get_python_include(repository_ctx, python_bin)
	File "/var/local/git/grpc/third_party/py/python_configure.bzl", line 253, column 13, in _get_python_include
		_execute(
	File "/var/local/git/grpc/third_party/py/python_configure.bzl", line 62, column 14, in _execute
		_fail("\n".join([
	File "/var/local/git/grpc/third_party/py/python_configure.bzl", line 35, column 9, in _fail
		fail("%sPython Configuration Error:%s %s\n" % (red, no_color, msg))
Error in fail: Python Configuration Error: Unable to find Python headers for /usr/bin/python2
```

https://source.cloud.google.com/results/invocations/f19bb81c-db82-474b-a2f7-910a9efd0ba4/log

--- a/bazel/internal_python_rules.bzl
+++ b/bazel/internal_python_rules.bzl
@@ -23,7 +23,7 @@ def internal_py_grpc_test(name, **kwargs):
       **kwargs: Any additional arguments to add to the test.
     """
     native.py_test(
-        name = name + ".natvie",
+        name = name + ".native",
         python_version = "PY3",
         **kwargs
     )

--- a/bazel/internal_python_rules.bzl
+++ b/bazel/internal_python_rules.bzl
@@ -14,7 +14,6 @@
 """Python-related rules intended only for use internal to the repo."""
 
 load("//bazel:gevent_test.bzl", "py_grpc_gevent_test")
-load("//bazel:python_rules.bzl", "py2and3_test")
 
 def internal_py_grpc_test(name, **kwargs):
     """Runs a test under all supported environments.
@@ -23,7 +22,11 @@ def internal_py_grpc_test(name, **kwargs):
       name: The name of the test.
       **kwargs: Any additional arguments to add to the test.
     """
-    py2and3_test(name, **kwargs)
+    native.py_test(
+        name = name + ".natvie",
+        python_version = "PY3",
+        **kwargs
+    )
     py_grpc_gevent_test(name, **kwargs)
 
     suite_kwargs = {}
@@ -33,7 +36,7 @@ def internal_py_grpc_test(name, **kwargs):
     native.test_suite(
         name = name,
         tests = [
-            name + ".both_pythons",
+            name + ".native",
             name + ".gevent",
         ],
         **suite_kwargs

--- a/bazel/python_rules.bzl
+++ b/bazel/python_rules.bzl
@@ -292,38 +292,3 @@ def py_grpc_library(
         strip_prefixes = strip_prefixes,
         **kwargs
     )
-
-# TODO(https://github.com/grpc/grpc/issues/27543): Remove once Python 2 is no longer supported.
-def py2and3_test(
-        name,
-        py_test = native.py_test,
-        **kwargs):
-    """Runs a Python test under both Python 2 and Python 3.
-
-    Args:
-      name: The name of the test.
-      py_test: The rule to use for each test.
-      **kwargs: Keyword arguments passed directly to the underlying py_test
-        rule.
-    """
-    if "python_version" in kwargs:
-        fail("Cannot specify 'python_version' in py2and3_test.")
-
-    names = [name + suffix for suffix in (".python2", ".python3")]
-    python_versions = ["PY2", "PY3"]
-    for case_name, python_version in zip(names, python_versions):
-        py_test(
-            name = case_name,
-            python_version = python_version,
-            **kwargs
-        )
-
-    suite_kwargs = {}
-    if "visibility" in kwargs:
-        suite_kwargs["visibility"] = kwargs["visibility"]
-
-    native.test_suite(
-        name = name + ".both_pythons",
-        tests = names,
-        **suite_kwargs
-    )

--- a/src/python/grpcio_tests/tests/unit/_cython/BUILD.bazel
+++ b/src/python/grpcio_tests/tests/unit/_cython/BUILD.bazel
@@ -12,8 +12,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-load("//bazel:python_rules.bzl", "py2and3_test")
-
 package(default_visibility = ["//visibility:public"])
 
 GRPCIO_TESTS_UNIT_CYTHON = [
@@ -37,7 +35,7 @@ py_library(
 )
 
 [
-    py2and3_test(
+    py_test(
         name = test_file_name[:-3],
         size = "small",
         srcs = [test_file_name],
@@ -46,6 +44,7 @@ py_library(
         ],
         imports = ["../../../"],
         main = test_file_name,
+        python_version = "PY3",
         deps = [
             ":common",
             ":test_utilities",

--- a/src/python/grpcio_tests/tests/unit/framework/foundation/BUILD.bazel
+++ b/src/python/grpcio_tests/tests/unit/framework/foundation/BUILD.bazel
@@ -1,5 +1,3 @@
-load("//bazel:python_rules.bzl", "py2and3_test")
-
 # Copyright 2021 The gRPC Authors
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
@@ -20,11 +18,12 @@ py_library(
     srcs = ["stream_testing.py"],
 )
 
-py2and3_test(
+py_test(
     name = "logging_pool_test",
     size = "small",
     srcs = ["_logging_pool_test.py"],
     main = "_logging_pool_test.py",
+    python_version = "PY3",
     deps = [
         "//src/python/grpcio/grpc:grpcio",
     ],

